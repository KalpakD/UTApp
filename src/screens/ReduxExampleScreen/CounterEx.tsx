import {increment} from 'actions/counterActionCreators';
import React, {useState} from 'react';
import {Button, Text, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';

const CounterEx = () => {
  const dispatch = useDispatch();
  const count = useSelector<any>(state => state.counter.count);

  return (
    <View style={{flex: 1}}>
      <Text style={{alignSelf: 'center', fontSize: 20}}>{count + ''}</Text>
      <Button
        testID="incrementBtn"
        title="Increment Couter"
        onPress={() => {
          //from Action Creator
          dispatch(increment(1));
        }}
      />
      <Button
        testID="decrementBtn"
        title="Decrement Couter"
        onPress={() => {
          // direct payload (without action creator)
          dispatch({
            type: 'DECREMENT',
            params: {diff: 1},
          });
        }}
      />
    </View>
  );
};
export default CounterEx;
