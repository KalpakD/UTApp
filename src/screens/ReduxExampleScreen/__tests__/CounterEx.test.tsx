/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import CounterEx from '../CounterEx';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
const mockStore = configureMockStore();

describe('Render Home Components correctly ', () => {
  let props = {};
  let initialState = {counter: {count: 0}};
  const defaultStore = mockStore(initialState);
  let wrapper: any;
  beforeEach(() => {
    wrapper = (props: any, store: any) => (
      <Provider store={store}>
        <CounterEx />
      </Provider>
    );
    render(wrapper(props, defaultStore));
  });

  afterEach(() => {
    jest.clearAllMocks();
    wrapper = null;
  });
  //this is  Snapshot testing
  it('renders correctly', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
  test('on Press Events on CounterEx Screen method 1', () => {
    const {getByTestId} = render(wrapper(props, defaultStore));
    fireEvent(getByTestId('incrementBtn'), 'onPress');
  });
  test('on Press Events on CounterEx Screen method 2', () => {
    fireEvent(screen.getByTestId('decrementBtn'), 'onPress');
  });
});
