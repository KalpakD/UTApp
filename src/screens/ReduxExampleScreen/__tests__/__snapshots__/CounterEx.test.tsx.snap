// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Components correctly  renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 20,
      }
    }
  >
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="incrementBtn"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment Couter
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="decrementBtn"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement Couter
      </Text>
    </View>
  </View>
</View>
`;
